[5.1.6] build failure on x32

Updated the sabyenc package to 5.1.6 yesterday, and after all archs on the ppa build fine I also did an upload to debian experimental (full list of builds [here](https://buildd.debian.org/status/package.php?p=python-sabyenc%2Csysbench&suite=experimental); some archs are still pending at the time of writing).

So far, only [x32](https://wiki.debian.org/X32Port) failed to build with some kind of assembler errors in crcutil:
```
x86_64-linux-gnux32-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -specs=/usr/share/dpkg/pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DCRCUTIL_USE_MM_CRC32=0 -Isrc/crcutil-1.0/code -Isrc/crcutil-1.0/tests -I/usr/include/python3.9 -c src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc -o /<<PKGBUILDDIR>>/.pybuild/cpython3_3.9_sabyenc3/src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.o -Wall -Wextra -Wno-unused-function -fomit-frame-pointer -fno-rtti -fno-exceptions -O3 -fPIC -fwrapv -std=c++11 -Wno-expansion-to-defined -Wno-unused-parameter
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc: Assembler messages:
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:144: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:145: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:155: Error: incorrect register `%edi' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:161: Error: `(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:163: Error: `(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:165: Error: `(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:167: Error: `(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:170: Error: `1*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:173: Error: `1*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:176: Error: `1*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:179: Error: `1*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:181: Error: `2*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:183: Error: `2*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:185: Error: `2*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:187: Error: `2*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:190: Error: `3*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:193: Error: `3*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:196: Error: `3*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:199: Error: `3*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:201: Error: `4*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:203: Error: `4*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:205: Error: `4*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:207: Error: `4*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:210: Error: `5*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:213: Error: `5*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:216: Error: `5*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:219: Error: `5*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:221: Error: `6*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:223: Error: `6*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:225: Error: `6*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:227: Error: `6*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:229: Error: `7*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:232: Error: `7*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:235: Error: `7*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:238: Error: `7*256*8(%r9d,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:240: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:244: Error: `(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:247: Error: `1*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:249: Error: `2*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:252: Error: `3*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:254: Error: `4*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:257: Error: `5*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:259: Error: `6*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:261: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:266: Error: `(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:269: Error: `1*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:271: Error: `2*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:274: Error: `3*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:276: Error: `4*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:279: Error: `5*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:281: Error: `6*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:283: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:288: Error: `(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:291: Error: `1*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:293: Error: `2*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:296: Error: `3*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:298: Error: `4*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:301: Error: `5*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:303: Error: `6*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:305: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:310: Error: `(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:313: Error: `1*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:315: Error: `2*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:318: Error: `3*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:320: Error: `4*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:323: Error: `5*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:325: Error: `6*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:327: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:328: Error: incorrect register `%edi' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:330: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:331: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:335: Error: incorrect register `%edi' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:338: Error: `(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:341: Error: `1*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:343: Error: `2*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:346: Error: `3*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:348: Error: `4*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:351: Error: `5*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:353: Error: `6*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:355: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:356: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:359: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:360: Error: incorrect register `%r11d' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:367: Error: incorrect register `%edi' used with `q' suffix
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:368: Error: `7*256*8(%ebp,%rsi,8)' is not a valid base/index expression
src/crcutil-1.0/code/multiword_64_64_gcc_amd64_asm.cc:369: Error: incorrect register `%r11d' used with `q' suffix
error: command '/usr/bin/x86_64-linux-gnux32-gcc' failed with exit code 1
```

Full build log attached.
[python-sabyenc_x32_5.1.6-1_stamp1648648018.log](https://github.com/sabnzbd/sabyenc/files/8391693/python-sabyenc_x32_5.1.6-1_stamp1648648018.log)

--- a/setup.py
+++ b/setup.py
@@ -35,17 +35,22 @@ from setuptools.command.build_ext import build_ext
 from distutils import log
 
 
+def log_info(inp: str):
+    """Convenience function"""
+    log.info("==> %s", inp)
+
+
 def autoconf_check(
     compiler: Type[CCompiler], include_check: str = None, define_check: str = None, flag_check: str = None
 ):
     """A makeshift Python version of the autoconf checks"""
     with tempfile.NamedTemporaryFile("w", suffix=".cc") as f:
         if include_check:
-            log.info("Checking support for include: %s", include_check)
+            log_info("Checking support for include: %s", include_check)
             f.write(f"#include <{include_check}>\n")
 
         if define_check:
-            log.info("Checking support for define: %s", define_check)
+            log_info("Checking support for define: %s", define_check)
             # Just let it crash
             f.write(f"#ifndef {define_check}\n")
             f.write(f"#error {define_check} not available!\n")
@@ -53,7 +58,7 @@ def autoconf_check(
 
         extra_postargs = []
         if flag_check:
-            log.info("Checking support for flag: %s", flag_check)
+            log_info("Checking support for flag: %s", flag_check)
             extra_postargs.append(flag_check)
 
         f.write("int main (int argc, char **argv) { return 0; }")
@@ -63,9 +68,9 @@ def autoconf_check(
 
         try:
             result_files = compiler.compile([f.name], extra_postargs=extra_postargs)
-            log.info("==> Success!")
+            log_info("Success!")
         except CompileError:
-            log.info("==> Not available!")
+            log_info("Not available!")
             return False
 
         # Remove output file(s)
@@ -83,7 +88,7 @@ class SAByEncBuild(build_ext):
         IS_ARM = machine.startswith("arm") or machine.startswith("aarch64")
         IS_AARCH64 = True
 
-        log.info("Baseline detection: ARM=%s, x86=%s, macOS=%s", IS_ARM, IS_X86, IS_MACOS)
+        log_info("Baseline detection: ARM=%s, x86=%s, macOS=%s", IS_ARM, IS_X86, IS_MACOS)
 
         # Determine compiler flags
         gcc_arm_neon_flags = []
@@ -117,7 +122,7 @@ class SAByEncBuild(build_ext):
                 cflags.append("-std=c++0x")
                 ext.extra_compile_args.append("-std=c++0x")
             else:
-                log.info("C++11 flag not available")
+                log_info("C++11 flag not available")
 
             # Verify specific flags for ARM chips
             # macOS M1 do not need any flags, they support everything
@@ -125,17 +130,28 @@ class SAByEncBuild(build_ext):
                 if not autoconf_check(self.compiler, include_check="sys/auxv.h"):
                     # We only tested this for GCC, might still be valid on MS-ARM compiler (see #38)
                     if autoconf_check(self.compiler, define_check="__GNUC__"):
-                        log.info("On GGC and sys/auxv.h not available, setting UNSUPPORTED_PLATFORM_ARM")
+                        log_info("On GGC and sys/auxv.h not available, setting UNSUPPORTED_PLATFORM_ARM=1")
                         ext.define_macros.append(("UNSUPPORTED_PLATFORM_ARM", "1"))
                         gcc_macros.append(("UNSUPPORTED_PLATFORM_ARM", "1"))
                         IS_ARM = False
                 if not autoconf_check(self.compiler, define_check="__aarch64__"):
-                    log.info("__aarch64__ not available, disabling 64bit extensions")
+                    log_info("__aarch64__ not available, disabling 64bit extensions")
                     IS_AARCH64 = False
                 if autoconf_check(self.compiler, flag_check="-march=armv8-a+crc"):
                     gcc_arm_crc_flags.append("-march=armv8-a+crc")
                 if autoconf_check(self.compiler, flag_check="-mfpu=neon"):
                     gcc_arm_neon_flags.append("-mfpu=neon")
+
+            # Check for special x32 case
+            if (
+                IS_X86 
+                and not IS_MACOS 
+                and autoconf_check(self.compiler, define_check="__ILP32__")
+                and autoconf_check(self.compiler, define_check="__x86_64__")
+            ):
+                log_info("Detected x32 platform, setting CRCUTIL_USE_ASM=0")
+                ext.define_macros.append(("CRCUTIL_USE_ASM", "0"))
+                gcc_macros.append(("CRCUTIL_USE_ASM", "0"))
 
         srcdeps_crc_common = ["src/yencode/common.h", "src/yencode/crc_common.h", "src/yencode/crc.h"]
         srcdeps_dec_common = ["src/yencode/common.h", "src/yencode/decoder_common.h", "src/yencode/decoder.h"]

--- a/setup.py
+++ b/setup.py
@@ -35,22 +35,17 @@ from setuptools.command.build_ext import build_ext
 from distutils import log
 
 
-def log_info(inp: str):
-    """Convenience function"""
-    log.info("==> %s", inp)
-
-
 def autoconf_check(
     compiler: Type[CCompiler], include_check: str = None, define_check: str = None, flag_check: str = None
 ):
     """A makeshift Python version of the autoconf checks"""
     with tempfile.NamedTemporaryFile("w", suffix=".cc") as f:
         if include_check:
-            log_info("Checking support for include: %s", include_check)
+            log.info("==> Checking support for include: %s", include_check)
             f.write(f"#include <{include_check}>\n")
 
         if define_check:
-            log_info("Checking support for define: %s", define_check)
+            log.info("==> Checking support for define: %s", define_check)
             # Just let it crash
             f.write(f"#ifndef {define_check}\n")
             f.write(f"#error {define_check} not available!\n")
@@ -58,7 +53,7 @@ def autoconf_check(
 
         extra_postargs = []
         if flag_check:
-            log_info("Checking support for flag: %s", flag_check)
+            log.info("==> Checking support for flag: %s", flag_check)
             extra_postargs.append(flag_check)
 
         f.write("int main (int argc, char **argv) { return 0; }")
@@ -68,9 +63,9 @@ def autoconf_check(
 
         try:
             result_files = compiler.compile([f.name], extra_postargs=extra_postargs)
-            log_info("Success!")
+            log.info("==> Success!")
         except CompileError:
-            log_info("Not available!")
+            log.info("==> Not available!")
             return False
 
         # Remove output file(s)
@@ -88,7 +83,7 @@ class SAByEncBuild(build_ext):
         IS_ARM = machine.startswith("arm") or machine.startswith("aarch64")
         IS_AARCH64 = True
 
-        log_info("Baseline detection: ARM=%s, x86=%s, macOS=%s", IS_ARM, IS_X86, IS_MACOS)
+        log.info("==> Baseline detection: ARM=%s, x86=%s, macOS=%s", IS_ARM, IS_X86, IS_MACOS)
 
         # Determine compiler flags
         gcc_arm_neon_flags = []
@@ -122,7 +117,7 @@ class SAByEncBuild(build_ext):
                 cflags.append("-std=c++0x")
                 ext.extra_compile_args.append("-std=c++0x")
             else:
-                log_info("C++11 flag not available")
+                log.info("==> C++11 flag not available")
 
             # Verify specific flags for ARM chips
             # macOS M1 do not need any flags, they support everything
@@ -130,12 +125,12 @@ class SAByEncBuild(build_ext):
                 if not autoconf_check(self.compiler, include_check="sys/auxv.h"):
                     # We only tested this for GCC, might still be valid on MS-ARM compiler (see #38)
                     if autoconf_check(self.compiler, define_check="__GNUC__"):
-                        log_info("On GGC and sys/auxv.h not available, setting UNSUPPORTED_PLATFORM_ARM=1")
+                        log.info("==> On GGC and sys/auxv.h not available, setting UNSUPPORTED_PLATFORM_ARM=1")
                         ext.define_macros.append(("UNSUPPORTED_PLATFORM_ARM", "1"))
                         gcc_macros.append(("UNSUPPORTED_PLATFORM_ARM", "1"))
                         IS_ARM = False
                 if not autoconf_check(self.compiler, define_check="__aarch64__"):
-                    log_info("__aarch64__ not available, disabling 64bit extensions")
+                    log.info("==> __aarch64__ not available, disabling 64bit extensions")
                     IS_AARCH64 = False
                 if autoconf_check(self.compiler, flag_check="-march=armv8-a+crc"):
                     gcc_arm_crc_flags.append("-march=armv8-a+crc")
@@ -149,7 +144,7 @@ class SAByEncBuild(build_ext):
                 and autoconf_check(self.compiler, define_check="__ILP32__")
                 and autoconf_check(self.compiler, define_check="__x86_64__")
             ):
-                log_info("Detected x32 platform, setting CRCUTIL_USE_ASM=0")
+                log.info("==> Detected x32 platform, setting CRCUTIL_USE_ASM=0")
                 ext.define_macros.append(("CRCUTIL_USE_ASM", "0"))
                 gcc_macros.append(("CRCUTIL_USE_ASM", "0"))
 

